### ISSUE TITLE
List item inside code annotation not rendering correctly

### ISSUE THREAD

**Issue by bnigh** [squidfunk/mkdocs-material#5633]
### Bug description

When including multiple annotations for a code block and an annotation contains a nested list, then the annotations do not render correctly.

The one annotation is rendering with the second unordered list item blank, and another annotation's content is replaced by the text of that missing list item.

This seems to be related to #5496, although is impacting code annotations for public version

### Related links

- [Code block annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations)
- #5496

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11842966/example.zip)

### Steps to reproduce

MkDocs version: 1.4.3
Material version: 9.1.16

- Follow stops for including [code blocks annotations](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations)

- Add markdown similar to the following

```md
    ```bash
    echo "foo" # (1)!
    echo "bar" # (2)!
    ```
    
    1. The beginning of the first annotation, which includes a:
	    
	    - list
	    - of
	    - items
	    
	    Plus another sentence.
	    
    2. Here's the second annotation.

```

And they render like

![Screen Shot 2023-06-22 at 7 41 44 PM](https://github.com/squidfunk/mkdocs-material/assets/8219313/a61e25dd-469b-4760-9a4d-883d5709036a)

![Screen Shot 2023-06-22 at 7 47 11 PM](https://github.com/squidfunk/mkdocs-material/assets/8219313/ed3e84d5-c204-4534-99d7-f8885bfee46d)




### Browser

_No response_

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in 3be797ece. Indeed, the fix should've been applied to the community edition as well. My fault.

**Comment by squidfunk** [comment]
Released as part of [9.1.17](https://github.com/squidfunk/mkdocs-material/releases/tag/9.1.17).

**Comment by bnigh** [comment]
Can confirm this resolves the issue I saw. Thank you (especially for such quick turn-around)!
